dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -331,6 +331,11 @@ protected void githubWebhookDelete(String repository, String gitReference, Strin
         workflows.forEach(workflow -> {
             workflow.getWorkflowVersions().removeIf(workflowVersion -> Objects.equals(workflowVersion.getName(), gitReferenceName.get()) && !workflowVersion.isFrozen());
             FileFormatHelper.updateEntryLevelFileFormats(workflow);
+
+            // Unpublish the workflow if it was published and no longer has any versions
+            if (workflow.getWorkflowVersions().isEmpty()) {
+                workflow.setIsPublished(false);
```

Comment:
Doesn't ES need to be updated as well? Or does that happen somewhere else?

Reply:
> Doesn't ES need to be updated as well? Or does that happen somewhere else?

We should probably update ES if any version was deleted, some of the indexed data might change based on the particulars of the remaining versions...